sdk/core/core-sse/test/util.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { assert } from "@azure/test-utils";
+
+export async function assertAsyncIterable<T>(
+  iter: AsyncIterable<T>,
+  count: number,
+  validate: (x: T) => void
+): Promise<void> {
+  let i = 0;
+  for await (const item of iter) {
+    validate(item);
```

Comment:
Could you pass in the index to the validate function so you don't have to use an outside indexer in some of your tests?